src/finding/mod.rs

```diff
@@ -524,6 +519,13 @@ impl<'w> FindingBuilder<'w> {
     }
 
     pub(crate) fn build(self, document: &'w impl AsRef<yamlpath::Document>) -> Result<Finding<'w>> {
+        self.build_with_document(document.as_ref())
+    }
+
+    pub(crate) fn build_with_document(
```

Comment:
Unfortunately there is this separate method needed now because apparently `&T` does not implement `AsRef<T>`. Not sure about this `build_with_document` document; suggestions are welcome.

Reply:
I thought about this a bit and added a new `AsDocument` trait -- this allows us to do what `AsRef` can't, since Rust won't let us do `impl<T: StepCommon> AsRef<str> for T` for coherence reasons.

AFAICT this gives us the best of both worlds: we don't need a `build`/`build_with_document` split _and_ we can have most things implement `AsDocument` to give us further flexibility.